lab_1_keywords_tfidf/main.py

```diff
@@ -99,6 +113,16 @@ def calculate_frequencies(tokens: list[str]) -> dict[str, int] | None:
         dict[str, int] | None: A dictionary {token: occurrences}.
         In case of corrupt input arguments, None is returned.
     """
+    if tokens is None:
+        return None
+    if not isinstance(tokens, list):
+        return None
+    frequency_dictionary = {}
+    for token in tokens:
+        if not isinstance(token, str):
```

Comment:
check firstly, then count. What if tokens amount will be 1000000000 and last one is incorrect, but you will spend a lot of time to wait for it and will have `None `.